src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,45 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_STORAGE];
+        values = (V[]) new Object[MAX_STORAGE];
+    }
+
+    public int getKeyPosition(K key) {
+        for (int i = 0; i < size; i++) {
+            if (key == keys[i] || key != null && key.equals(keys[i])) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
     @Override
     public void put(K key, V value) {
+        int keyPosition = getKeyPosition(key);
+        if (keyPosition != -1) {
+            keys[keyPosition] = key;
+            values[keyPosition] = value;
+        } else {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        return (getKeyPosition(key) == -1) ? null : values[getKeyPosition(key)];
```

Comment:
better save the result of `getKeyPosition(key)` to the separate variable
```suggestion
        int index = getKeyPosition(key);
        return (index == -1) ? null : values[index];
```